core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -918,19 +959,40 @@ private void endRequest0(@Nullable Throwable requestCause, long requestEndTimeNa
         }
 
         if (name == null) {
+            String newServiceName = null;
             String newName = null;
-            final RpcRequest rpcReq = ctx.rpcRequest();
+            RpcRequest rpcReq = ctx.rpcRequest();
+            if (rpcReq == null && requestContent instanceof RpcRequest) {
+               rpcReq = (RpcRequest) requestContent;
+            }
+
             if (rpcReq != null) {
+                newServiceName = rpcReq.serviceType().getName();
                 newName = rpcReq.method();
-            } else if (requestContent instanceof RpcRequest) {
-                newName = ((RpcRequest) requestContent).method();
-            }
+            } else if (ctx instanceof ServiceRequestContext) {
+                final ServiceConfig config = ((ServiceRequestContext) ctx).config();
+                newServiceName = config.defaultServiceName();
+                if (newServiceName == null) {
+                    newServiceName = config.route().meterTag();
+                }
 
-            if (newName == null && ctx instanceof ServiceRequestContext) {
-                newName = ((ServiceRequestContext) ctx).config().defaultLogName();
+                newName = config.defaultLogName();
+                if (newName == null) {
+                    newName = ctx.method().name();
+                }
             }
 
+            serviceName = newServiceName;
             name = newName;
+            if (serviceName != null) {
+                if (name != null) {
+                    fullName = serviceName + '/' + name;
+                } else {
+                    fullName = serviceName;
+                }
+            } else {
+                fullName = name;
+            }
```

Comment:
- Shouldn't this be out of `if (name == null)` block?
- We need to distinguish between 'name-only' and 'serviceName-only' cases.